### PR TITLE
Adding logging functionality

### PR DESCRIPTION
**Pull request by devagupt**

### CODE REVIEW

File: hubblestack/extmods/modules/nova.py [hunk 1]
```diff
@@ -162,20 +181,23 @@ def __execute_module(audit_id, audit_impl, audit_data, verbose):
...
         module_result_local = __nova__[execute_method](audit_id, audit_check)
         audit_result_local = {}
         if module_result_local['result']:
             audit_result_local['check_result'] = CHECK_STATUS['Success']
         else:
-            # if 'failure_reason' in audit_data:
-            #     audit_result_local
             audit_result_local['check_result'] = CHECK_STATUS['Failure']
-        
+            audit_result_local['failure_reason'] = module_result_local['failure_reason']
```
Comment (goravsingal) [inline hubblestack/extmods/modules/nova.py]: In profile schema, we have introduced this attribute: "failure_schema"
If it is present, that should override any failure reason value from modue.

File: hubblestack/extmods/modules/nova.py [hunk 2]
```diff
@@ -162,20 +181,23 @@ def __execute_module(audit_id, audit_impl, audit_data, verbose):
     for audit_check in audit_impl['checks']:
         __nova__[validate_param_method](audit_id, audit_check)
 
+    #check for type in check implementation. If not present default is 'and'
+    type = audit_impl.get('type', 'and').lower().strip()
+    audit_result['type']=type
```
Comment (goravsingal) [inline hubblestack/extmods/modules/nova.py]: spacing around '='

File: hubblestack/extmods/modules/nova.py [hunk 3]
```diff
@@ -187,12 +209,21 @@ def __execute_module(audit_id, audit_impl, audit_data, verbose):
...
-    log.info(json.dumps(audit_result, sort_keys=False, indent=4))
-    # log.info(audit_result)
-    log.info('~~~~~~~~~~~~~~~~~~~~~~~~')
+        if type == 'and':
+            overall_result = overall_result and module_result_local['result']
+        else:
+            overall_result = overall_result or module_result_local['result']
+
+    #Update overall check result
+    if overall_result:
```
Comment (goravsingal) [inline hubblestack/extmods/modules/nova.py]: These 4 lines can be written in 1 line using lambda epression

File: hubblestack/extmods/modules/nova.py [hunk 4]
```diff
@@ -187,12 +209,21 @@ def __execute_module(audit_id, audit_impl, audit_data, verbose):
...
+        audit_result['check_result'] = CHECK_STATUS['Success']
+    else:
+        audit_result['check_result'] = CHECK_STATUS['Failure']
+
+    # log.info('~~~~~~~ Result ~~~~~~~~~')
```
Comment (goravsingal) [inline hubblestack/extmods/modules/nova.py]: remove these comments

File: hubblestack/extmods/modules/nova.py [hunk 5]
```diff
@@ -336,3 +367,46 @@ def __get_audit_files(audit_files):
...
+    :param result_dict:
+    :param combined_dict:
+    :param show_compliance:
+    :return:
+    """
+    for audit_file in combined_dict:
+        result_list = combined_dict[audit_file]
+        for result in result_list:
+            dict={}
+            dict[result['tag']]=result['description']
```
Comment (goravsingal) [inline hubblestack/extmods/modules/nova.py]: I think you missed that attribute which says: "Do not include me in individual result". 
I think that is: "sub_check: True" or sub_module

File: profiles/test.yaml [hunk 6]
```diff
@@ -0,0 +1,318 @@
+check_unique_id:
```
Comment (goravsingal) [inline profiles/test.yaml]: I think you committed this by mistake. Currently, there is no profile stored in code.

File: hubblestack/extmods/modules/nova.py [hunk 7]
```diff
@@ -162,20 +181,23 @@ def __execute_module(audit_id, audit_impl, audit_data, verbose):
...
         __nova__[validate_param_method](audit_id, audit_check)
 
+    #check for type in check implementation. If not present default is 'and'
+    type = audit_impl.get('type', 'and').lower().strip()
+    audit_result['type']=type
     # validate succeded, lets execute it and prepare result dictionary
     audit_result['run_config']['checks'] = []
     execute_method = audit_impl['module'] + '.execute'
     filtered_log_method = audit_impl['module'] + '.get_filtered_params_to_log'
+    overall_result = type=='and'
```
Comment (goravsingal) [inline hubblestack/extmods/modules/nova.py]: You can put some comment about the usage of this variable.

**Review by goravsingal — CHANGES_REQUESTED** [review]

File: profiles/test.yaml (hunk 6 above)
Comment (devagupt) [reply profiles/test.yaml]: As discussed in last demo, a sample profile is required to be stored. I assumed it to be stored alongside code.

Comment (devagupt) [reply profiles/test.yaml]: will remove it from code.

File: hubblestack/extmods/modules/nova.py (hunk 1 above)
Comment (devagupt) [reply hubblestack/extmods/modules/nova.py]: Done

File: hubblestack/extmods/modules/nova.py (hunk 5 above)
Comment (devagupt) [reply hubblestack/extmods/modules/nova.py]: Done

File: hubblestack/extmods/modules/audit.py [hunk 8]
```diff
@@ -1,450 +1,544 @@
...
+            result_list.append(skipped_dict)
+            log.error(version_error)
         except Exception as exc:
-            log.exception('Error loading audit file %s: %s', audit_file, exc)
-            return False, audit_data
-    if not audit_data or not isinstance(audit_data, dict):
-        log.error('audit data from %s was not formed as a dict', audit_file)
-        return False, audit_data
-    return True, audit_data
+            import traceback
```
Comment (goravsingal) [inline hubblestack/extmods/modules/audit.py]: This is not needed, remove stacktrace

File: profiles/test.yaml (hunk 6 above)
Comment (goravsingal) [reply profiles/test.yaml]: This is still here

**Review by goravsingal — CHANGES_REQUESTED** [review]

File: hubblestack/extmods/modules/audit.py [hunk 9]
```diff
@@ -1,450 +1,544 @@
...
-            version_match = True
-
-    return version_match
-
-
-def _run_audit(ret, audit_data, tags, labels, audit_file):
+    for audit_file in combined_dict:
+        result_list = combined_dict[audit_file]
+        for result in result_list:
+            sub_check = result.get('sub_check', False)
```
Comment (goravsingal) [inline hubblestack/extmods/modules/audit.py:439]: You are ignoring results after this, this is needed during boolean expression evaluation later.

File: hubblestack/extmods/modules/audit.py (hunk 8 above)
Comment (devagupt) [reply hubblestack/extmods/modules/audit.py]: Removed

File: hubblestack/extmods/modules/audit.py (hunk 9 above)
Comment (devagupt) [reply hubblestack/extmods/modules/audit.py:439]: Boolean expressions will be evaluated in this code here. The logic needs to be modified here only when boolean expression is going to be written.

**Pull request merged**